public/js/admin/users.js

```diff
@@ -86,22 +86,44 @@ var createUsers = {
     },
     validateRolesAndPassword: function () {
         var pass = true;
-        //validate password
+        // Adding password constraint!
+        var passwordErrors = "";
+        var password = $.trim(document.forms["createForm"]["password"].value);
+        if(password.length < 6){
+            passwordErrors += "password is less than 6 characters ";
+            //createUsers.elements.password.next(".errors").text("password is less than 6 characters\n");
+            pass = false;
+        }
+        var re = /[A-Z]/;
+        if(!re.test(password)){
+            passwordErrors += "password must have at least one upper case ";
+            //createUsers.elements.password.next(".errors").text("password must have at least one upper case\n");
+            pass = false;
+        }
+        re = /[0-9]/;
         if ($.trim(document.forms["createForm"]["password"].value) === "") {
-            createUsers.elements.password.next(".errors").text("please assign this user a password");
+            passwordErrors = "please assign this user a password";
             pass = false;
-        } else {
+        }
+        else if(!re.test(password)) {
+            passwordErrors +=  "password does not have a number ";
+            // createUsers.elements.password.next(".errors").text("password does not have a number\n");
+            pass = false;
+        }
+
+        if(pass == false)
+        {
+                createUsers.elements.password.next (".errors").text(passwordErrors);
+        }
+        else{
             createUsers.elements.password.next(".errors").text("");
         }
+
         //validate roles
-        var isARoleChecked = false;
-        $.each(document.forms["createForm"].elements["roles[]"], function () {
-            if ($(this).is(':checked')) {
-                isARoleChecked = true;
-            }
-        });
-        if (isARoleChecked === false) {
+        if (typeof document.forms["createForm"].elements["roles[]"] === 'undefined') {
             pass = false;
+        }
+        if (pass === false) {
```

Comment:
AJ,

It looks like the logic for validating if a role is checked has some flaws. I don't think the foreach loop (previously line 98-103) can be removed. Try the following steps to see what I am referring to:

1) Submit a new user with all valid fields, except no roles checked. The validation still passes and hits the server, which catches the error instead of catching it in JavaScript.

2) Submit a new user with no fields filled out. After the errors appear, select at least one role and click submit. The 'select at least one role' error no longer disappears.